scoring/tests/test_scoring.py

```diff
@@ -51,6 +52,73 @@ def test_template(self):
             "Should return score values for every team",
         )
 
+    def test_no_cans(self):
+        self.assertScores(
+            {'ABC': 0, 'DEF': 0},
+            {0: {'tokens': ""}, 1: {'tokens': ""}},
+        )
+
+    def test_cans(self):
+        self.assertScores(
+            {'ABC': 0, 'DEF': 4},
+            {0: {'tokens': "S"}, 1: {'tokens': "TBS"}},
+        )
+
+    def test_cans_and_left_scoring_zone(self):
+        self.teams_data['ABC']['left_scoring_zone'] = True
+        self.assertScores(
+            {'ABC': 1, 'DEF': 4},
+            {0: {'tokens': "S"}, 1: {'tokens': "TBS"}},
+        )
+
+    def test_invalid_can_characters(self):
```

Comment:
For these validation tests it would be great if they could also exercise cases where the invalid parts are in zones which are not occupied. I think the implementation does handle this correctly, so it should just be a matter of tweaking the tests.

Reply:
Does [3249d81](https://github.com/srobo/sr2022-comp/pull/2/commits/3249d818bd3bd44f3de445f3b0b1bfa6cb6fc6e4) cover what you were thinking?